mcs/class/corlib/Test/System.Threading/ThreadPoolTest.cs

```diff
@@ -255,5 +255,29 @@ public void AsyncLocalCapture ()
 			Assert.AreEqual (0, var_3, "var_3");
 		}
 
+		[Test]
+		public static void PerformanceCounter ()
+		// This is test related to bug https://bugzilla.xamarin.com/show_bug.cgi?id=41294.
+		// The bug is that the performance counters return 0.
+		// "Work Items Added" and "# of Threads" are fixed, the others are not.
+		{
+			var workItems = new PerformanceCounter ("Mono Threadpool", "Work Items Added");
+			var threads   = new PerformanceCounter ("Mono Threadpool", "# of Threads");
+
+			var workItems0 = workItems.NextValue();
+
+			int N = 99;
+			for (var i = 0; i < N; i++)
+				ThreadPool.QueueUserWorkItem (_ => {});
+
+			var workItems1 = workItems.NextValue();
+			var threads0 = threads.NextValue();
+
+			Console.WriteLine ("workItems0:{0} workItems1:{1}", workItems0, workItems1);
+			Console.WriteLine ("{0}:{1}", "threads", threads0);
+
+			Assert.IsTrue (workItems1 - workItems0 == N, "#1");
+			Assert.IsTrue (threads0 > 0, "#2");
```

Comment:
There is no guarantee the ThreadPool is going to launch a new thread, so asserting on it is flaky.

Reply:
Can items be run directly on invoking thread, i.e. on lightly loaded system, not worth creating a thread, or always queued to a separate thread? I think always queued.

Reply:
See https://github.com/mono/mono/pull/6118#pullrequestreview-80534889